### PR TITLE
Fix block to poll dynamic locations

### PR DESCRIPTION
**Pull request by kyliedale**

### CODE REVIEW

File: weather_underground_base.py [hunk 1]
```diff
@@ -40,59 +41,37 @@ class WeatherUndergroundBase(RESTPolling):
...
-    def _process_response(self, resp):
-        """ Extract weather conditions from response.
+    def process_signals(self, signals):
+        response = None
+        weather_signals = []
 
-        Args:
-            resp (Response)
+        for signal in signals:
+            for locations in self.queries():
```
Comment (mattdodge) [inline weather_underground_base.py]: I don't think we need (nor should) `queries` to be a list anymore. Maybe just make a `city` and `state` prop and use those.

File: weather_underground_base.py [hunk 2]
```diff
@@ -15,7 +16,7 @@ class Location(PropertyHolder):
 
 
 @not_discoverable
-class WeatherUndergroundBase(RESTPolling):
+class WeatherUndergroundBase(Block):
```
Comment (mattdodge) [inline weather_underground_base.py]: We'll want to use the Retry mixin here too, that can be a different PR though

File: weather_underground_base.py [hunk 3]
```diff
@@ -40,59 +41,37 @@ class WeatherUndergroundBase(RESTPolling):
...
-
-    @property
-    def current_query(self):
-        return self.queries()[self._idx]
+                                          state,
+                                          city)
+
+        try:
+            resp = requests.get(self.url, headers=headers)
+        except Exception as e:
```
Comment (mattdodge) [inline weather_underground_base.py]: requests rarely raises an exception, only if something bad happens. If an error code comes back it is included in the `resp` object. So in our exception catch here we definitely want to log to error:
```python
except:
    self.logger.exception("GET request failed")
```

In addition, inside the `try` we should check that the `resp.status` is what we expect (200, probably) and if it is not log an error. Once the retry mixin is in place, we'll want to raise an exception in that case instead so that the request will retry.

File: weather_underground_base.py (hunk 1 above)
Comment (kyliedale) [inline weather_underground_base.py]: Should you only be able to poll for one city and state at a time?

Comment (mattdodge) [inline weather_underground_base.py]: Yeah. If we want multiple cities/states just use multiple blocks. This will become important when we get into retry loops and stuff like that. Keeping track of different city/state combo statuses is tricky and it's way easier to just throw another block next to it.

**Comment by mattdodge** [comment]
Will make a separate issue to add the retry mixin to this block, this looks good for now.
